clues.go

```diff
@@ -95,6 +81,24 @@ func marshal(a any) string {
 	return string(bs)
 }
 
+type cluesCtxKey struct{}
```

Comment:
Odd diff.  This code is unchanged.  I actually moved `marshal` up above this code.